src/client/Microsoft.Identity.Client/Cache/SuggestedWebCacheKeyFactory.cs

```diff
@@ -54,6 +55,13 @@ private static bool GetOboOrAppKey(AuthenticationRequestParameters requestParame
 
             if (requestParameters.ApiId == TelemetryCore.Internal.Events.ApiEvent.ApiIds.AcquireTokenForClient)
             {
+                if (!string.IsNullOrEmpty(requestParameters.Authority.TenantId) &&
+                    Guid.TryParse(requestParameters.Authority.TenantId, out var tenantId))
```

Comment:
Change is fine with one small issue: tenantId may not be a GUID, for example "bgavril.onmicrosoft.com" is a valid tenant ID. 
We only care if tenatnID is not null or empty.

Reply:
Yeah I wasn't sure about that.
It also means that tenantid can be "common" or "organizations, etc. 
I'm guessing that's ok? It'll result in a cache miss for those who already have those tokens cached.